dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -62,6 +65,89 @@ protected WorkflowResource(final WorkflowHelper workflowHelper,
         this.workflowAPI    = workflowAPI;
     }
 
+    /**
+     * Returns all schemes non-archived. 401 if the user does not have permission.
+     * @param request  HttpServletRequest
+     * @return Response
+     */
+    @GET
+    @Path("/schemes")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response findSchemes(@Context final HttpServletRequest request,
+                                      @QueryParam("contentTypeId") final String contentTypeId) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
+        List<WorkflowScheme> schemes;
+
+        try {
+
+            Logger.debug(this,
+                    "Getting the workflow schemes for the contentTypeId: " + contentTypeId);
+            schemes   = (null != contentTypeId)?
+                    this.workflowHelper.findSchemesByContentType
+                            (contentTypeId, initDataObject.getUser()):
+                    this.workflowHelper.findSchemes();
+
+            response  =
+                    Response.ok(new ResponseEntityView(schemes)).build(); // 200
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)